src/EnergyPlus/General.cc

```diff
@@ -91,945 +91,6 @@ using DataHVACGlobals::Bisection;
-        }
-        case DataHVACGlobals::HVACSystemRootSolverAlgorithm::RegulaFalsiThenBisection: {
-            if (NIte > state.dataHVACGlobal->HVACSystemRootFinding.NumOfIter) {
-                XTemp = (X1 + X0) / 2.0;
-            } else {
-                XTemp = (Y0 * X1 - Y1 * X0) / DY;
-            }
-            break;
-        }
-        case DataHVACGlobals::HVACSystemRootSolverAlgorithm::BisectionThenRegulaFalsi: {
-            if (NIte <= state.dataHVACGlobal->HVACSystemRootFinding.NumOfIter) {
-                XTemp = (X1 + X0) / 2.0;
-            } else {
-                XTemp = (Y0 * X1 - Y1 * X0) / DY;
-            }
-            break;
-        }
-        case DataHVACGlobals::HVACSystemRootSolverAlgorithm::Alternation: {
-            if (AltIte > state.dataHVACGlobal->HVACSystemRootFinding.NumOfIter) {
-                XTemp = (X1 + X0) / 2.0;
-                if (AltIte >= 2 * state.dataHVACGlobal->HVACSystemRootFinding.NumOfIter) AltIte = 0;
-            } else {
-                XTemp = (Y0 * X1 - Y1 * X0) / DY;
-            }
-            break;
-        }
-        default: {
-            XTemp = (Y0 * X1 - Y1 * X0) / DY;
-        }
-        }
-
-        YTemp = f(state, XTemp, Par);
-
-        ++NIte;
-        ++AltIte;
-
-        // check convergence
-        if (std::abs(YTemp) < Eps) Conv = true;
-
-        if (NIte > MaxIte) StopMaxIte = true;
-
-        if ((!Conv) && (!StopMaxIte)) {
-            Cont = true;
-        } else {
-            Cont = false;
-        }
-
-        if (Cont) {
-
-            // reassign values (only if further iteration required)
-            if (Y0 < 0.0) {
-                if (YTemp < 0.0) {
-                    X0 = XTemp;
-                    Y0 = YTemp;
-                } else {
-                    X1 = XTemp;
-                    Y1 = YTemp;
-                }
-            } else {
-                if (YTemp < 0.0) {
-                    X1 = XTemp;
-                    Y1 = YTemp;
-                } else {
-                    X0 = XTemp;
-                    Y0 = YTemp;
-                }
-            } // ( Y0 < 0 )
-
-        } // (Cont)
-
-    } // Cont
-
-    if (Conv) {
-        Flag = NIte;
-    } else {
-        Flag = -1;
-    }
-    XRes = XTemp;
-}
-
-void SolveRoot(Real64 const Eps, // required absolute accuracy
-               int const MaxIte, // maximum number of allowed iterations
-               int &Flag,        // integer storing exit status
-               Real64 &XRes,     // value of x that solves f(x,Par) = 0
-               std::function<Real64(Real64 const, Array1D<Real64> const &)> f,
-               Real64 const X_0,           // 1st bound of interval that contains the solution
-               Real64 const X_1,           // 2nd bound of interval that contains the solution
-               Array1D<Real64> const &Par, // array with additional parameters used for function evaluation
-               int const AlgorithmTypeNum, // ALgorithm selection
```

Comment:
For places that were passing in `AlgorithmTypeNum` and that flavor of `SolveRoot` only supported bisection and regula falsi), do those components need some documentation changes?

Reply:
If I recall correctly, the only place I found it being used like that was actually _inside_ a unit test.  I could pull develop, delete that one overload and try to build to verify unless you happen to know that there are definitely components using it.